pytest-embedded-serial-esp/pytest_embedded_serial_esp/serial.py

```diff
@@ -101,38 +99,13 @@ def __init__(
             )
 
         with contextlib.redirect_stdout(msg_queue):
-            # Temp workaround for esptool
-            # on windows have to close the unused scanned ports manually
-            #
-            # could revert to the following code blocks after fixing it
-            #
-            # esp: esptool.ESPLoader = esptool.get_default_connected_device(
-            #     ports,
-            #     port=port,
-            #     connect_attempts=3,
-            #     initial_baud=baud,
-            #     chip=esptool_target,
-            # )
-            _esp = None
-            for each_port in reversed(ports):
-                print(f'Serial port {each_port}')
-                try:
-                    if esptool_target == 'auto':
-                        _esp = detect_chip(each_port, baud, connect_attempts=3)
-                    else:
-                        chip_class = CHIP_DEFS[esptool_target]
-                        _esp = chip_class(each_port, baud)
-                        _esp.connect(attempts=3)
-                    break
-                except (FatalError, OSError) as err:
-                    if port is not None:
-                        raise
-                    print(f'{each_port} failed to connect: {err}')
-                    if _esp:
-                        # ensure unused port is closed.
-                        _esp._port.close()
-                    _esp = None
-            esp = _esp
+            esp = esptool.get_default_connected_device(
```

Comment:
yeah, if we have `self.esp = ...` here, we don't have to `detect_chip` everytime in `use_esptool`.